src/Application/Service/CatalogDumpService.php

```diff
@@ -2,90 +2,88 @@
 
 declare(strict_types=1);
 
-namespace LinkedDataSets\Application\Job;
+namespace LinkedDataSets\Application\Service;
 
 use EasyRdf\Graph;
 use EasyRdf\RdfNamespace;
 use EasyRdf\Resource;
 use Laminas\Log\Logger;
+use Laminas\Log\LoggerInterface;
 use Laminas\ServiceManager\Exception\ServiceNotFoundException;
-use Laminas\ServiceManager\ServiceLocatorInterface;
-use Omeka\Entity\Job;
-use Omeka\Job\AbstractJob;
-use Omeka\Job\Exception\InvalidArgumentException;
 
-final class CatalogDumpJob extends AbstractJob
+final class CatalogDumpService
 {
-    // in php 8.1 convert to enum
-    const DUMP_FORMATS = [
+    private const DUMP_FORMATS = [
         "turtle" => "ttl",
         "ntriples" => "nt",
         "jsonld" => "jsonld",
         "rdfxml" => "xml",
     ];
 
     protected ?Logger $logger = null;
-    protected $serverUrl;
+    protected $uriHelper;
     protected $id;
 
-    public function __construct(Job $job, ServiceLocatorInterface $serviceLocator)
-    {
-        parent::__construct($job, $serviceLocator);
-        $this->logger = $serviceLocator->get('Omeka\Logger');
-        if (!$this->logger) {
-            throw new ServiceNotFoundException('The logger service is not found');
-        }
-        $this->serverUrl = $serviceLocator->get('ViewHelperManager')?->get('ServerUrl');
-        if (!$this->serverUrl) {
-            throw new ServiceNotFoundException('The serverUrl service is not found');
-        }
-        $this->id = $this->getArg('id');
-        if (!$this->id) {
-            throw new InvalidArgumentException('No catalog_id was provided to the job');
+    public function __construct(
+        LoggerInterface $logger,
+        $uriHelper
+    ) {
+        $this->logger = $logger;
+        $this->uriHelper = $uriHelper;
+        if (!$this->uriHelper) {
+            throw new ServiceNotFoundException('The UriHelper service is not found');
```

Comment:
In general, this construction can be covered by requiring a typed version of the helper class in the constructors parameter. Also by not allowing a null value. That way the error would be automatically triggered without you having to add logic to verify correct dependencies are set.